### PR TITLE
Categorize FAQ section

### PR DESCRIPTION
**Pull request by MTSaeed**
**What does this PR?**

Categorize the FAQ page. By using Accordion, makes the cards of the each category . 
By clicking on toggle icon of any category, it will open the card and that card displays items of that particular category.
By Searching anything from the search bar, it will hide the accordion and displays the table of content.

**Steps to reproduce**

1. Click on the FAQ or type /faq in the URL.
2. you will see the Accordion. Click on the toggle icon to expand accordion, it will show the content corresponding to that 
    particular category.
3. By clicking on any row of accordion body, it will redirect to that specific part.
4. When user searches anything from the search bar, it will hide the accordion and display the data in the form of  table of 
    content.

**Issue number** 

fixes #[1056](https://github.com/ecency/ecency-vision/issues/1056)

**Screenshot / Video**

https://user-images.githubusercontent.com/106739598/210234188-db7f1bcb-8d97-4c34-aa2b-8950f3a10630.mp4

### CODE REVIEW

File: src/common/components/faq-category/index.scss [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  padding: 1rem;
+}
+
+.section-container {
+  @include themify(night) {
+        background: $dark;
+  }
+}
+
+.section-content:hover{
```
Comment (dkildar) [inline src/common/components/faq-category/index.scss:24]: Missed space before `{`

File: src/common/components/faq-category/index.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  background: #dee2e6;
```
Comment (dkildar) [inline src/common/components/faq-category/index.scss:25]: Should it be themed?

File: src/common/components/faq-category/index.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import Accordion from "react-bootstrap/Accordion";
+import { Button, Card, OverlayTrigger, Tooltip } from "react-bootstrap";
+
+import { chevronDownSvgForSlider, chevronUpSvgForSlider } from "../../img/svg";
+
+import { _t } from "../../i18n";
+
+interface Props {
+  categoryTitle: string;
+  contentList: Array<string>;
```
Comment (dkildar) [inline src/common/components/faq-category/index.tsx:11]: Just `string[]` better

File: src/common/components/faq-category/index.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+const FaqCategory = (props: Props) => {
+  const [expanded, setExpanded] = useState(false);
+  let tooltip = (
```
Comment (dkildar) [inline src/common/components/faq-category/index.tsx:16]: Its `const` I guess

File: src/common/components/faq-category/index.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    <Tooltip id="profile-tooltip1" style={{ zIndex: 10 }}>
```
Comment (dkildar) [inline src/common/components/faq-category/index.tsx:17]: Need to rename ID for some meaningful

File: src/common/components/faq-category/index.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                </div>
+              </Accordion.Toggle>
+            </OverlayTrigger>
+          </div>
+        </div>
+        <Accordion.Collapse eventKey="0">
+          <Card.Body className="p-0">
+            <div>
+              {contentList.map((x) => {
+                return (
```
Comment (dkildar) [inline src/common/components/faq-category/index.tsx:51]: You may remove `{ return` use like: `() => <>...</>`

File: src/common/pages/_static.scss [hunk 7]
```diff
@@ -452,3 +452,15 @@
 .text-not-found{
   font-size: 16px;
 }
+.category-cards{
```
Comment (dkildar) [inline src/common/pages/_static.scss:455]: Space before `{`

File: src/common/pages/_static.scss [hunk 8]
```diff
@@ -452,3 +452,15 @@
...
+  margin-bottom: 6%;
```
Comment (dkildar) [inline src/common/pages/_static.scss:456]: Same as issue above

File: src/common/pages/_static.scss [hunk 9]
```diff
@@ -452,3 +452,15 @@
...
+  border-radius: 6px;
+  @include themify(day) {
+    border: 1px solid #dee2e6;
+  }
+
+  @include themify(night) {
+    border: 1px  solid $dark;
```
Comment (dkildar) [inline src/common/pages/_static.scss:463]: Extra space after `1px`

File: src/common/pages/static/faq.tsx [hunk 10]
```diff
@@ -101,7 +102,7 @@ class FaqPage extends Component<PageProps, FAQPageState> {
           itemType="https://schema.org/FAQPage"
         >
           <div className="static-content">
-            <div className="position-relative rounded">
+            <div className="position-relative rounded" style={{ marginBottom: "8%" }}>
```
Comment (dkildar) [inline src/common/pages/static/faq.tsx:105]: Same as issue above

**Comment by MTSaeed** [comment]
Updated.

File: src/common/pages/static/faq.tsx [hunk 11]
```diff
@@ -150,16 +151,38 @@ class FaqPage extends Component<PageProps, FAQPageState> {
...
+                      <li key={x}>
+                        <a href={`#${x}`}>{_t(`static.faq.${x}-header`)}</a>
+                      </li>
+                    );
+                  })}
+                </ul>
+              </>
+            ) : (
+              <div className="category-cards">
+                <FaqCategory categoryTitle={"About Ecency"} contentList={dataToShow.slice(0, 5)} />
```
Comment (feruzm) [inline src/common/pages/static/faq.tsx:170]: @MTSaeed  These titles should be translated as well.

File: src/common/i18n/locales/en-US.json [hunk 12]
```diff
@@ -1424,6 +1424,7 @@
       "contact-email": "Email us"
     },
     "faq": {
+      "toggle-icon-info": "click me",
```
Comment (feruzm) [inline src/common/i18n/locales/en-US.json:1427]: Change text to "Tap on me to toggle"

File: src/common/components/faq-category/index.tsx [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+                </div>
+              </Accordion.Toggle>
+            </OverlayTrigger>
+          </div>
+        </div>
+        <Accordion.Collapse eventKey="0">
+          <Card.Body className="p-0">
+            <div>
+              {contentList.map((x) => (
+                <a className="section-content" href={`#${x}`}>
```
Comment (feruzm) [inline src/common/components/faq-category/index.tsx:51]: need to add prop `key={x}` for uniqueness.

**Review by feruzm — APPROVED** [review]
lgtm

**Pull request merged**